Sources/Classes/RSDBPersistentManager.m

```diff
@@ -176,6 +144,105 @@ - (int)openEncryptedDB:(NSString *)encryptionKey {
     return executeCode;
 }
 
+- (void)handleWhenEncryptedDBExists:(RSDBEncryption * _Nullable)dbEncryption isEncryptionNeeded:(BOOL)isEncryptionNeeded {
+    if (isEncryptionNeeded) {
+        // open encrypted database with key
+        int code = [self openEncryptedDB:dbEncryption.key];
+        if (code == SQLITE_NOTADB) {
+            // when key is wrong
+            // delete encrypted database; then open new encrypted database
+            // all previous events will be deleted
+            [RSLogger logError:@"RSDBPersistentManager: createDB: Wrong key is provided. Deleting previous encrypted DB and creating a new encrypted DB"];
+            [self closeDB];
+            [RSUtils removeFile:ENCRYPTED_DB_NAME];
+            [self openEncryptedDB:dbEncryption.key];
+        }
+        return;
+    }
+    
+    // when encryption is disabled and no key is provided
+    // delete encrypted database; then open unencrypted database
+    // all previous events will be deleted
+    if (dbEncryption == nil || dbEncryption.key == nil) {
+        [RSLogger logError:@"RSDBPersistentManager: createDB: No key is provided. Deleting encrypted DB and creating a new unencrypted DB"];
+        [RSUtils removeFile:ENCRYPTED_DB_NAME];
+        [self openUnencryptedDB];
+        return;
+    }
```

Comment:
Should we consider moving this block of code at the start of the current method? And also add the same check in `handleWhenUnencryptedDBExists`?
Reason: It would ensure that the `key` is valid before attempting to perform any operation on the DB with the `key`.

Reply:
We already checking if the key is valid or not within the `isEncryptionNeeded` method and the output of it was used before encrypting the DB with the key. 

This was the only case where we were using the key for a purpose other than encryption, hence we are doing the check here to see if the key is valid or not before using it.

However improved it further by moving this logic into a separate method to check if a valid key is supplied by the user or not